### PR TITLE
boards/arm/efr32xg24_dk2601b: add I2C support

### PR DESCRIPTION
**Pull request by fkokosinski**
This PR adds I2C support to the efr32xg24_dk2601b board. It uses the pre-existing Gecko I2C driver.

### CODE REVIEW

File: boards/arm/efr32xg24_dk2601b/efr32xg24_dk2601b.dts [hunk 1]
```diff
@@ -39,6 +40,9 @@
 		blue_led: led_1 {
 			gpios = <&gpiob GECKO_PIN(0) GPIO_ACTIVE_LOW>;
 		};
+		sensorenable: sensorenable {
+			gpios = <&gpioc GECKO_PIN(9) GPIO_ACTIVE_HIGH>;
+		};
```
Comment (MaureenHelm) [inline boards/arm/efr32xg24_dk2601b/efr32xg24_dk2601b.dts:45]: Unrelated change?

Comment (fkokosinski) [reply boards/arm/efr32xg24_dk2601b/efr32xg24_dk2601b.dts:45]: This pin is used for enabling the on-board i2c sensors, which are hidden behind a switch, i.e. you have to enable this pin with the Zephyr GPIO API to use scan/read/write on them.

Should I remove it anyway?

Comment (MaureenHelm) [reply boards/arm/efr32xg24_dk2601b/efr32xg24_dk2601b.dts:45]: Yes. Please look at using `regulator-fixed` for this

**Comment by fkokosinski** [comment]
Hi @MaureenHelm. I've addressed your comments - could you PTAL at this PR?

**Review by MaureenHelm — APPROVED** [review]

**Review by tgorochowik — APPROVED** [review]
